pkg/kube/pod.go

```diff
@@ -64,12 +64,11 @@ type PodOptions struct {
 	OwnerReferences    []metav1.OwnerReference
 }
 
-// CreatePod creates a pod with a single container based on the specified image
-func CreatePod(ctx context.Context, cli kubernetes.Interface, opts *PodOptions) (*v1.Pod, error) {
+func GetPodObjectFromPodOptions(cli kubernetes.Interface, opts *PodOptions) (*v1.Pod, string, error) {
 	// If Namespace is not specified, use the controller Namespace.
 	cns, err := GetControllerNamespace()
 	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to get controller namespace")
+		return nil, "", errors.Wrapf(err, "Failed to get controller namespace")
 	}
 	ns := opts.Namespace
```

Comment:
Can we move this check to the caller `CreatePod` so we don't have to return it from this function?

Reply:
Yes. The latest commit addresses this comment 👍🏼